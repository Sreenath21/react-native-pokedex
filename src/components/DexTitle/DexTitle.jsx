import React from 'react';
import {View, Text, Pressable} from 'react-native';
import {CommonActions} from '@react-navigation/native';

import {getZeroFilledId} from '../../utils/helplerFunctions';
import styles from './DexTitle.style';

const DexTitle = ({name, id, navigation}) => {
  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>{name.toUpperCase()}</Text>
        <Text style={styles.id}>{getZeroFilledId(id)}</Text>
      </View>

      <Pressable
        style={styles.btnContainer}
        // onPress={() => navigation.navigate('Home')}>
        onPress={() => navigation.dispatch(CommonActions.goBack())}>
        <Text style={styles.closeButton}>x</Text>
      </Pressable>
    </View>
  );
};

export default DexTitle;
