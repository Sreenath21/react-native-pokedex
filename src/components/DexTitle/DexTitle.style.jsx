import {StyleSheet} from 'react-native';
import colors from '../../utils/colors';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'space-between',
    marginTop: 32,
  },
  titleContainer: {},
  title: {
    fontSize: 32,
    fontWeight: 'bold',
    color: colors.primary,
    letterSpacing: 1.2,
  },
  id: {
    fontSize: 32,
    fontWeight: 400,
    color: colors.primary,
  },
  btnContainer: {
    justifyContent: 'flex-start',
    alignSelf: 'flex-start',
    paddingHorizontal: 8,
    borderWidth: 1,
    borderColor: colors.primary,
    borderRadius: 50,
  },
  closeButton: {
    fontSize: 24,
    color: colors.primary,
    textAlign: 'center',
  },
});
