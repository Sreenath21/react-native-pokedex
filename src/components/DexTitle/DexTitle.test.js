import React from 'react';
import DexTitle from './DexTitle';
import renderer from 'react-test-renderer';

it('Checking DexTitle component is rendering', () => {
  const tree = renderer.create(
    <DexTitle name="bulbasaur" id={1} navigation={jest.fn()} />,
  );
  expect(tree).toMatchSnapshot();
});
