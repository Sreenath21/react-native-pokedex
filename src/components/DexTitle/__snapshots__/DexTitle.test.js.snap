// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking DexTitle component is rendering 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 32,
    }
  }
>
  <View
    style={{}}
  >
    <Text
      style={
        {
          "color": "#391363",
          "fontSize": 32,
          "fontWeight": "bold",
          "letterSpacing": 1.2,
        }
      }
    >
      BULBASAUR
    </Text>
    <Text
      style={
        {
          "color": "#391363",
          "fontSize": 32,
          "fontWeight": 400,
        }
      }
    >
      001
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "flex-start",
        "borderColor": "#391363",
        "borderRadius": 50,
        "borderWidth": 1,
        "justifyContent": "flex-start",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#391363",
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      x
    </Text>
  </View>
</View>
`;
